tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -133,6 +134,111 @@ public TypeMetadataHelper()
             }
         }
 
+        public void CompareMethodMetadata(
+            CmdletMetadata cmdlet,
+            TypeMetadata oldType,
+            TypeMetadata newType,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            var oldMethods = oldType.Methods;
+            var newMethods = newType.Methods;
+
+            foreach (var oldMethod in oldMethods)
+            {
+                bool found = false;
+                var candidateMethods = newMethods.Where(m => string.Equals(m.Name, oldMethod.Name)).ToList();
+                foreach (var newMethod in candidateMethods)
+                {
+                    var oldParameters = oldMethod.Parameters;
+                    var newParameters = newMethod.Parameters;
+                    if (oldParameters.Count != newParameters.Count)
+                    {
+                        continue;
+                    }
+
+                    bool match = true;
+                    for (int idx = 0; idx < oldParameters.Count; idx++)
+                    {
+                        var oldParameter = oldParameters[idx];
+                        var newParameter = newParameters[idx];
+                        if (oldParameter.Name != newParameter.Name || oldParameter.Type != newParameter.Type)
+                        {
+                            match = false;
+                            break;
+                        }
+                    }
+
+                    if (match)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: cmdlet,
+                        severity: 0,
+                        problemId: 0,
+                        description: string.Empty,
+                        remediation: string.Empty);
+                }
+            }
+        }
+
+        public void CompareConstructorMetadata(
+            CmdletMetadata cmdlet,
+            TypeMetadata oldType,
+            TypeMetadata newType,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            var oldConstructors = oldType.Constructors;
```

Comment:
Other than return types, seems like we should be able to reuse the same code here.

Reply:
Added a `MethodSignature` class that `MethodMetadata` and `ConstructorMetadata` inherit from and refactored these two methods into one to compare method signatures.